irods/session.py

```diff
@@ -67,6 +67,51 @@ def available_permissions(self):
             self.__access = _iRODSAccess_pre_4_3_0 if self.server_version < (4,3) else iRODSAccess
         return self.__access
 
+    # TODO - write tests for new interface: sessionObject.groups
+    @property
+    def groups(self):
+        class GroupsManager(self.user_groups.__class__):
+
+            def create(self, name,
+			     user_type = 'rodsgroup',
```

Comment:
I wasn't really participating in the discussion about the interface the other day, so, sorry if this is rehashing something...

Is the `GroupsManager` responsible for managing groups, or for filling a sort of `groupadmin` set of capabilities (including creating users)?

If it's the former, I would think that the only type of `user_type` we would allow is `rodsgroup` and we shouldn't give a choice here (harder to hold it wrong). This would also remove the need for a `user_zone` since there's no such thing as remote groups and `auth_str` since groups do not authenticate.

If it's the latter, is `session.groups` conveying well that this is for managing users *and* groups? I've definitely been alone in being confused by things like this in the past, so if this makes sense to everyone else, please just resolve this. :)

Reply:
`GroupManager`, like the older `UserGroupManager`, is purely for managing groups, while users are managed and created with `UserManager`.  I was wondering about `user_type`, `user_zone`, and `auth_str` myself, but probably won't make a move on removing those until it's explained to me why they might have been included in the API in the first place.

Reply:
> I would think that the only type of user_type we would allow is `rodsgroup` and we shouldn't give a choice here (harder to hold it wrong). This would also remove the need for a `user_zone` since there's no such thing as remote groups and `auth_str` since groups do not authenticate.

I've put in another commit that removes these parameters from the groups interface. I guess we'll leave it in the user_groups interface (now deprecated, probably to be removed some in the next few releases), just to do the least damage possible to existing code bases.

Reply:
(If I should make an issue corresponding the removal of those parameters, and that's significant enough to be its own commit in the end, let me know.)

Reply:
No, let's not touch the existing interface (`user_groups`). I was just suggesting the change for the new thing (`groups`... if indeed it is new). I think what you have in your latest changes looks very nice!

Reply:
This being the case, I think that this can be resolved.